arch/arc/include/asm/pgalloc.h

```diff
@@ -88,9 +88,22 @@ static inline void p4d_populate(struct mm_struct *mm, p4d_t *p4dp, pud_t *pudp)
 
 static inline pud_t *pud_alloc_one(struct mm_struct *mm, unsigned long addr)
 {
+	pud_t *pud;
+
 	BUILD_BUG_ON((PTRS_PER_PUD * sizeof(pud_t)) > PAGE_SIZE);
 
-	return (pud_t *)__get_free_page(GFP_KERNEL | __GFP_ZERO);
+	/*
+	 * For kernel in high address, there will be a dedicated swapper_pud
+	 * and its base pointet needs to be set in user pgd
+	 */
+	BUILD_BUG_ON(PAGE_OFFSET != 0x80000000);
+
+	pud = (pud_t *)__get_free_page(GFP_KERNEL | __GFP_ZERO);
+
+	pud[2] = swapper_pud[2];
```

Comment:
It will be nice to have something like `pud_idx(PAGE_OFFSET)` and `pud_idx(PAGE_OFFSET+PUD_SIZE)` here just for understanding why this numbers are used.